src/ec/packages/ec-component-hero-banner/hero-banner.html.twig

```diff
@@ -0,0 +1,78 @@
+{#
+  Parameters:
+    - "type" (string) (default: 'default') Type of banner (can be 'default','image','image-shade','primary')
+    - "title" (string) (default: '') Title of banner
+    - "image" (string) (default: '') Image for banner (required for image banner type)
+    - "description" (string) (default: '') Description of banner
+    - "centered" (bool) (default: true) Define if banner should be centered
+    - "button" (associative array) (default: predefined structure) predefined structure for EC Button component
+    - "extra_classes" (string) (default: '')
+    - "extra_attributes" (array) (default: []): format: [
+        {
+          "name" (string) (default: ''),
+          "value" (string) (default: '')
+        },
+        ...
+      ]
+#}
+
+{# variables #}
+{% set _type = type|default('default') %}
+{% set _centered = centered|default(true) %}
+{% set _title = title|default('') %}
+{% set _description = description|default('') %}
+{% set _image = image|default('') %}
+
+{% set _button = {
+  variant: 'call',
+  icon: {
+    type: 'ui',
+    name: 'corner-arrow',
+    path: '',
+    size: 'xs',
+    transform: 'rotate-90'
+  },
+  label: '',
+  extra_classes: 'ecl-hero-banner__button'
+  }
+%}
+
+{% set _css_class = ['ecl-hero-banner ecl-hero-banner--'~_type]|join(' ') %}
+{% set _extra_attributes = '' %}
+
+{# Internal logic - Process properties #}
+{% if button is defined and button is not empty %}
+  {% set _button_icon = _button.icon %}
+  {% set _button = _button|merge(button) %}
+  {% set _button = _button|merge({ icon: _button_icon|merge(button.icon) }) %}
+{% endif %}
+
+{% if _centered == true %}
+  {% set _css_class=_css_class~' ecl-hero-banner--centered' %}
+{% endif %}
+
+{% if extra_classes is defined and extra_classes is not empty %}
+  {% set _css_class = _css_class ~ ' ' ~ extra_classes %}
+{% endif %}
+
+{% if extra_attributes is defined and extra_attributes is not empty and extra_attributes is iterable %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %}
+
+{# Print the result #}
+{% spaceless %}
```

Comment:
As you may have seen on slack, we had a lots of discussion concerning the best way to handle whitespaces.
We ended up with a [set of rules](https://github.com/ec-europa/ecl-twig/blob/5aadb4708da7e5fe27d7bbb7b8b1d91fae4adacc/docs/decisions/001-strip-whitespaces.md).
Please have a look at it.
Basically the only changes that have to be done here is to put the spaceless at the top of the file, and remove dashes for inline twig tags

Reply:
According to the documentation, now (https://github.com/ec-europa/ecl-twig/pull/49/commits/b61ae00e3ae1e87c10ce6b1bddf1950e0fae7785) it should be good - all tags placed inside (and not between) should be removed `-`, but where I have conditional instructions or loading another component, the `-` should stay?